From 6826e5d31d6323eac5137404f0194bf2183b561c Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Wed, 7 Nov 2018 16:48:47 +0100
Subject: [PATCH] Add device TCTI library to the initramfs

The tpm2-tools don't dynamically link against the TCTI libraries anymore,
but instead dlopen() the correct library depending on the TCTI used.

So dracut isn't able anymore to figure out automatically using ldd what
libraries are needed by the tpm2-tools. Since clevis uses the device TCTI
to access the TPM directly, add the libtss2-tcti-device.so to the initrd.

Suggested-by: Federico Chiacchiaretta <federico.chia@gmail.com>

Fixes: ##74
---
 src/luks/systemd/dracut/module-setup.sh.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/luks/systemd/dracut/module-setup.sh.in b/src/luks/systemd/dracut/module-setup.sh.in
index 41e7d6c..990bf4a 100755
--- a/src/luks/systemd/dracut/module-setup.sh.in
+++ b/src/luks/systemd/dracut/module-setup.sh.in
@@ -65,6 +65,7 @@ install() {
 	    tpm2_pcrlist \
 	    tpm2_unseal \
 	    tpm2_load
+	inst_libdir_file "libtss2-tcti-device.so*"
     fi
 
     dracut_need_initqueue
-- 
2.27.0

